numba/core/types/npytypes.py

```diff
@@ -647,3 +647,15 @@ def __init__(self, coef, domain=None, window=None, n_args=1):
         # constructor, since the types of domain and window arguments depend on
         # that and we need that information when boxing
         self.n_args = n_args
+
+
+class ChebyshevType(Type):
+    def __init__(self, coef, domain=None, window=None, n_args=1):
+        super(ChebyshevType, self).__init__(name=f'ChebyshevType({coef}, {domain}, {domain}, {n_args})')
```

Comment:
You can reuse most of the code written for Polynomial here. Create a base type, move all the code common to both types to the base type and inherit from base instead of `Type.`